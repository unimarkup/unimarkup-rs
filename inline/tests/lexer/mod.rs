use crate::Snapshot;
use unimarkup_commons::test_runner::{self, snap_test_runner::SnapTestRunner};
use unimarkup_inline::Tokenize;

mod snapshot;

#[test]
fn test_lexer_snapshots() {
    for case in crate::prepare_test_cases("spec/markup", "spec/snapshots/lexer") {
        let symbols = test_runner::scan_str(&case.input);
        let runner = SnapTestRunner::with_fn(&case.name, &symbols, |symbols| {
            let rest = &[];
            let snapshot = Snapshot::snap((case.input.as_ref(), symbols.tokens()));
            (snapshot, rest)
        })
        .with_info(format!("Test '{}' from '{}'", case.name, case.file_name));

        unimarkup_commons::run_snap_test!(runner, &case.out_path);
    }
}
