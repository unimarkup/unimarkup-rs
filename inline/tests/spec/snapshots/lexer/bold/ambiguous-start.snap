---
source: inline/tests/lexer/mod.rs
info: "Test 'ambiguous-start' from 'markup/bold.yml'"
---
The next ***word** is bold.
The
^^^ -> Plain @ (1:1)->(1:4)
    
   ^ -> Whitespace @ (1:4)->(1:5)
    next
    ^^^^ -> Plain @ (1:5)->(1:9)
         
        ^ -> Whitespace @ (1:9)->(1:10)
         *
         ^ -> Plain @ (1:10)->(1:11)
          **
          ^^ -> Bold @ (1:11)->(1:13)
            word
            ^^^^ -> Plain @ (1:13)->(1:17)
                **
                ^^ -> Bold @ (1:17)->(1:19)
                   
                  ^ -> Whitespace @ (1:19)->(1:20)
                   is
                   ^^ -> Plain @ (1:20)->(1:22)
                      
                     ^ -> Whitespace @ (1:22)->(1:23)
                      bold.
                      ^^^^^ -> Plain @ (1:23)->(1:28)
                           ⏎
                           ^ -> Newline @ (1:28)->(2:1)

---
With input:

The next ***word** is bold.


